src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -324,7 +409,7 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 
 		if (Array.isArray<string>(storedRecommendationsJson)) {
 			for (let id of <string[]>storedRecommendationsJson) {
-				if (allRecommendations.indexOf(id) > -1) {
+				if (allRecommendations.indexOf(id) > -1 && this.isExtensionAllowedToBeRecommended(id)) {
```

Comment:
`allRecommendations` and `this._availableRecommendations` shouldnt have the ignored recommendations in the first place.